javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/Issue3859Test.java

```diff
@@ -0,0 +1,59 @@
+package com.github.javaparser.symbolsolver;
+
+import com.github.javaparser.ParserConfiguration;
+import com.github.javaparser.StaticJavaParser;
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.expr.LambdaExpr;
+import com.github.javaparser.ast.expr.NameExpr;
+import com.github.javaparser.symbolsolver.resolution.AbstractResolutionTest;
+import com.github.javaparser.symbolsolver.resolution.typesolvers.CombinedTypeSolver;
+import com.github.javaparser.symbolsolver.resolution.typesolvers.ReflectionTypeSolver;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class Issue3859Test extends AbstractResolutionTest {
+
+    @Test
+    void test() {
+        String code = "import java.util.function.Consumer;\n" +
+                "\n" +
+                "class Demo {\n" +
+                "    void foo(Consumer<String> arg) {}\n" +
+                "    void print(Object arg) {}\n" +
+                "    public void bar() {\n" +
+                "        foo(s->print(s));\n" +
+                "    }\n" +
+                "    public void baz() {\n" +
+                "        foo((s->print(s)));\n" +
+                "    }\n" +
+                "}\n";
+        ParserConfiguration configuration = new ParserConfiguration()
```

Comment:
For some time this can be written more simply
> CompilationUnit cu = JavaParserAdapter.of(createParserWithResolver(defaultTypeSolver())).parse(code);

Reply:
Nice, I didn't know that.

Will change accordingly